src/simplay-web/test/EventFactory.integration.test.ts

```diff
@@ -0,0 +1,194 @@
+import { EventFactory } from '../src/EventFactory';
+import { EventSerialized } from '../src/EventSerialized';
+import { expect } from 'chai';
+import { SetVisibleEvent } from '../src/SetVisibleEvent';
+import { EventAction } from '../src/EventAction';
+import { SetPositionEvent } from '../src/SetPositionEvent';
+import { SetInteractingEvent } from '../src/SetInteractingEvent';
+import { SetNotInteractingEvent } from '../src/SetNotInteractingEvent';
+import { MoveNearEvent } from '../src/MoveNearEvent';
+import { MoveNearCellEvent } from '../src/MoveNearCellEvent';
+import { SetTintColorEvent } from '../src/SetTintColorEvent';
+import { SetDecoratingTextEvent } from '../src/SetDecoratingTextEvent';
+import { SetSpriteFrameEvent } from '../src/SetSpriteFrameEvent';
+import { ResourceSetCapacityEvent } from '../src/ResourceSetCapacityEvent';
+import { ResourceSetUtilizationEvent } from '../src/ResourceSetUtilizationEvent';
+import { ContainerSetCapacityEvent } from '../src/ContainerSetCapacityEvent';
+import { ContainerSetLevelEvent } from '../src/ContainerSetLevelEvent';
+import { StoreSetCapacityEvent } from '../src/StoreSetCapacityEvent';
+import { StoreSetContentEvent } from '../src/StoreSetContentEvent';
+
+const forId = 'leet';
+const timestamp = 1337;
+
+describe('EventFactory tests', function () {
+  it('should initialize SetVisibleEvent', () => {
+    const event = EventFactory.fromSerialized({
+      action: EventAction.SET_VISIBLE,
+      forId: forId,
+      timestamp: timestamp,
+      args: { visible: true },
+    } as EventSerialized) as SetVisibleEvent;
+    expect(event instanceof SetVisibleEvent).to.be.true;
```

Comment:
Can you use `.to.be.an.instanceof(...)` (https://www.chaijs.com/api/bdd/#method_instanceof) instead, for all of expect calls in this file